firmware/src/main.cpp

```diff
@@ -104,7 +119,7 @@ void update_app_state()
         app_regs.pressure_temp_humidity[2] = data.humidity_prh;
         
         // Send an event for the aggregate register
-        if (!HarpCore::is_muted() && bool(app_regs.enable_sensor_dispatch_events))
+        if (!HarpCore::is_muted() && HarpCore::events_enabled() && bool(app_regs.enable_sensor_dispatch_events))
```

Comment:
It's critical that we check that events are enabled before dispatching events. Otherwise, controllers may not deterministically connect to the device.

I'll probably need to draft out a few more protocol "gotchas" in a tutorial somewhere, but at least we have reviews for now.